query/encode.go

```diff
@@ -166,6 +173,8 @@ func reflectValue(values url.Values, val reflect.Value) error {
 				del = ','
 			} else if opts.Contains("space") {
 				del = ' '
+			} else if opts.Contains("key") {
```

Comment:
> for me, the confusing thing about names like repeatkey or multikey is that it sort of implies that without that option the field isn't repeated, when it actually is.

true, it has been so long since the PR I forgot :)

> What was the original motivation for this anyway? I'm guessing it's to handle servers in PHP (and other languages?) that don't know how to properly parse a=1&a=2 and erroneously have later values overwrite previous ones?

Yes, PHP, Rails and others use `[]` at the end of param names to specify the repeated parameter is an string.

Is there a spec about what's the proper way to parse query parameters with repeated names?

> Regarding confusion alongside comma and space (those being used to specify delimiters, and this being used to specify param suffix), I think we have to deal with the regardless. I don't think something like [] or brackets is any more confusing than anything else... we'll always have to rely on users of the library reading the docs to understand what these options do.

Happy to go with whichever you like the most. Should we go with `brackets` then?